### PR TITLE
Using the CustomHW in the HardwareInterface to implement the GeneralIO Module. Fixes #2 and #3

### LINKED ISSUE #2: Implement GeneralIO in ronex_hardware_interface

### PR DESCRIPTION
**Pull request by ugocupcic**

### CODE REVIEW

File: sr_ronex_ethercat_drivers/src/sr_board_mk2_gio.cpp [hunk 1]
```diff
@@ -163,6 +159,17 @@ int SrBoardMk2GIO::initialize(pr2_hardware_interface::HardwareInterface *hw, boo
 
   device_offset_ = sh_->get_ring_position();// - hand_->getBridgeRingPosition();
 
+  //add the RoNeX to the hw interface
+  general_io_.reset( new ronex::GeneralIO() );
+  general_io_->name_ = device_name_;
```
Comment (markpitchless) [inline sr_ronex_ethercat_drivers/src/sr_board_mk2_gio.cpp:164]: There seems to be lots of access to the member vars of GeneralIO outside of its class. Like here accessing the name, but also command_ etc. This feels a little wrong to me when I see the underscore on the end of the name. I feel like they should either have accessor methods and be private member data or not be named with a trailing underscore to indicate they are public member data of the class.
But maybe I'm mis-reading C++ conventions I'm not used to?

Comment (ugocupcic) [inline sr_ronex_ethercat_drivers/src/sr_board_mk2_gio.cpp:164]: No I find it weird too but it's the way they're doing it in pr2 hardware interface....

Comment (markpitchless) [inline sr_ronex_ethercat_drivers/src/sr_board_mk2_gio.cpp:164]: Ok, that makes sense, following the existing convention.

Comment (ugocupcic) [inline sr_ronex_ethercat_drivers/src/sr_board_mk2_gio.cpp:164]: just as a note, this is NOT our own convention so don't start doing that in
the rest of the code.

On 27 June 2013 15:25, Mark Pitchless notifications@github.com wrote:

> In sr_ronex_ethercat_drivers/src/sr_board_mk2_gio.cpp:
> 
> > @@ -163,6 +159,17 @@ int SrBoardMk2GIO::initialize(pr2_hardware_interface::HardwareInterface *hw, boo
> > 
> >    device_offset_ = sh_->get_ring_position();// - hand_->getBridgeRingPosition();
> > -  //add the RoNeX to the hw interface
> > -  general_io_.reset( new ronex::GeneralIO() );
> > -  general_io_->name_ = device_name_;
> 
> Ok, that makes sense, following the existing convention.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/shadow-robot/sr-ronex-dev/pull/7/files#r4915392
> .

## 

Ugo Cupcic | Shadow Robot Company | ugo@shadowrobot.com
Senior Software Engineer | 251 Liverpool Road |
need a Hand? | London N1 1LX | +44 20 7700 2487
http://www.shadowrobot.com/hand/ @shadowrobot

**Pull request merged**